### ISSUE TITLE
Get Player X and Y pos

### ISSUE THREAD

**Issue by Sonicmaster13** [dimotsai/agar-mini-map#53]
Greetings, 
  This is not an issue exactly, but their is no other way of contacting you (that i know....). I am creating a script of my own (a userscript), and it is going great, but I have run into an issue that has frozen me in my tracks, and I just can't fix it! I need to be able to get the players current X and Y, and the mouse' X and Y values, and then log it to localStorage.x and localStorage.y,  localStorage.Mx, and finally localStorage.My to use later, then put that on loop so it always has the most up to date X and Y values. I am pretty sure the X and Y are Float32 packets, and the mouse in Float64. I have tried many times to actually get this to work, but it looks as if your script can do it (i can see it can, it shows the X and Y at the bottom right with the minimap). Any help at all is appreciated!! (ps: if you want other way of contacting me i will provide my email, and i will close this 'issue' immediatly. Thanks!)

**Comment by dimotsai** [comment]
Could you show me how you get the position? It would be great if you provide the code that you are developing.

**Comment by Sonicmaster13** [comment]
Ok, i would rather not give out my script (yet), but i basically saw that your mini map was able to show the players X and Y value. I would like if you could help me develop my own script that can show the players X and Y values the same as yours, but only that alone, not any of the mini map stuff. All i need is the mechanic to get the players X and Y values, then draw that text on the screen, in the top left.

**Comment by dimotsai** [comment]
I don't know how much you already progress. But it would be helpful if you trace the following functions:

1. [extractPacket](https://github.com/dimotsai/agar-mini-map/blob/master/agar-mini-map.user.js#L737)
2. [extractCellPacket](https://github.com/dimotsai/agar-mini-map/blob/master/agar-mini-map.user.js#L613-L734)
3. [Cell](https://github.com/dimotsai/agar-mini-map/blob/master/agar-mini-map.user.js#L527-L587) and [Cell.updatePos](https://github.com/dimotsai/agar-mini-map/blob/master/agar-mini-map.user.js#L563-L586)

Basically, I replaced the vanilla `WebSocket` with my own to get the raw data from a server. Function `extractPacket` and function `extractCellPacket` are originally from official client code. In `extractPacket`, opcode 16 is to extract the information of all cells, and opcode 32 is to get the current  cell id numbers that belongs to you. I think those functions are enough to get what you want.

**Comment by Sonicmaster13** [comment]
Ok. I get what you said, but i'm not so sure i understand what you're code is saying. I am meh at coding in js, but I am not a pro like you (:D) i just finished learning the basics of javascript a few weeks ago and have been practicing ever sense. I have practically no script made so far, except for the functions you told me to look at ( and all of the other code for other numerous functions, but that's not on the point of getting the X and Y). I understand a bit about agarios' protocol, but not too much. Where would i put the commands i wanted, like console.log on loop for testing, and what variable would i use to do something like console.log on loop the X value? Also, if i just copy the functions you said, i get syntax errors when loading it in agario: 

    Syntax error @ 'RSAgar Packet Extract'!
    ##########################
    JSHINT output:
    script: (W030) Expected an assignment or function call and instead saw an expression on line: 56        at character: 50
    script: (E041) Unrecoverable syntax error. (33% scanned) on line: 84 at character: 1
    ##########################

    SyntaxError: missing ) after argument list
        at Function (native)
        at c (<anonymous>:40:425)
        at eval (eval at <anonymous> (unknown source), <anonymous>:1:25)
        at eval (native)
        at p (<anonymous>:71:373)
        at ha (<anonymous>:40:468)
        at ka (<anonymous>:82:289)
        at b (<anonymous>:83:12)

**Comment by dimotsai** [comment]
All X and Y values are stored in Cell objects: https://github.com/dimotsai/agar-mini-map/blob/master/agar-mini-map.user.js#L699-L700.

**Comment by Sonicmaster13** [comment]
Ok, i tried that and it did not report accurate x and y locations, according to Agario Powerups X and Y counter. I have another script that i (kinda) made that i have tried but get an undefined error when it IS defined... Here is a bit of it:  

   function Packet(e) {
        if(e instanceof Buffer) {
            this.data   = e;
            this.length = this.data.length;
        }else if((typeof Buffer) != 'undefined' && e.data instanceof Buffer) {
            this.data   = e.data;
            this.length = this.data.length;
        }else{
            this.data   = new DataView(e.data);
            this.length = this.data.byteLength;
        }

        this.offset = 0;
    }


    Packet.prototype = {
        readFloat64LE: function(p) {
            var offset = (typeof p) == 'number' ? p : this.offset;
            var ret;
            if(this.data.getFloat64) {
                ret = this.data.getFloat64(offset, true);
            }else{
                ret = this.data.readDoubleLE(offset);
            }
            if(p === undefined) this.offset += 8;

            return ret;
        }
    }

    var x    = Packet.readFloat32LE();
    console.log(x)
    var y    = Packet.readFloat32LE();
    console.log(y)

The error is here: ERROR: Execution of script 'Get X Y test' failed! Packet.readFloat32LE is not a function
